test/lib/url_language_switcher_test.rb

```diff
@@ -795,17 +795,28 @@ def test_add_lang_code_with_custom_domain_langs
         ['https://my-site.com/index.php?a=1&b=2#hash', 'zh-Hant-HK', 'https://zh-hant-hk.com/zh/index.php?a=1&b=2#hash'],
 
         # absolute path
-        ['/', 'en', '/'],
-        ['/', 'ja', '/ja/'],
-        ['/index.php', 'ja', '/ja/index.php'],
-        ['/a/b/', 'ja', '/ja/a/b/'],
-        ['/a/b/index.php', 'ja', '/ja/a/b/index.php'],
-        ['/index.php', 'en-US', '/index.php'],
-        ['/index.php', 'zh-CHS', '/zh/chs/index.php'],
-        ['/index.php', 'zh-Hant-HK', '/zh/index.php'],
-        ['/index.php?a=1&b=2', 'zh-Hant-HK', '/zh/index.php?a=1&b=2'],
-        ['/index.php#hash', 'zh-Hant-HK', '/zh/index.php#hash'],
-        ['/index.php?a=1&b=2#hash', 'zh-Hant-HK', '/zh/index.php?a=1&b=2#hash'],
+        ['/', 'en', 'http://my-site.com/'],
+        ['/', 'ja', 'http://my-site.com/ja/'],
+        ['/index.php', 'ja', 'http://my-site.com/ja/index.php'],
+        ['/a/b/', 'ja', 'http://my-site.com/ja/a/b/'],
+        ['/a/b/index.php', 'ja', 'http://my-site.com/ja/a/b/index.php'],
+        ['/index.php', 'en-US', 'http://en-us.my-site.com/index.php'],
+        ['/index.php', 'zh-CHS', 'http://my-site.com/zh/chs/index.php'],
+        ['/index.php', 'zh-Hant-HK', 'http://zh-hant-hk.com/zh/index.php'],
+        ['/index.php?a=1&b=2', 'zh-Hant-HK', 'http://zh-hant-hk.com/zh/index.php?a=1&b=2'],
+        ['/index.php#hash', 'zh-Hant-HK', 'http://zh-hant-hk.com/zh/index.php#hash'],
+        ['/index.php?a=1&b=2#hash', 'zh-Hant-HK', 'http://zh-hant-hk.com/zh/index.php?a=1&b=2#hash'],
+
+        # relative path
+        ['index.php', 'ja', 'http://my-site.com/ja/req_uri/index.php'],
+        ['a/b/', 'ja', 'http://my-site.com/ja/req_uri/a/b/'],
+        ['a/b/index.php', 'ja', 'http://my-site.com/ja/req_uri/a/b/index.php'],
+        ['index.php', 'en-US', 'http://en-us.my-site.com/req_uri/index.php'],
+        ['index.php', 'zh-CHS', 'http://my-site.com/zh/chs/req_uri/index.php'],
+        ['index.php', 'zh-Hant-HK', 'http://zh-hant-hk.com/zh/req_uri/index.php'],
+        ['index.php?a=1&b=2', 'zh-Hant-HK', 'http://zh-hant-hk.com/zh/req_uri/index.php?a=1&b=2'],
+        ['index.php#hash', 'zh-Hant-HK', 'http://zh-hant-hk.com/zh/req_uri/index.php#hash'],
+        ['index.php?a=1&b=2#hash', 'zh-Hant-HK', 'http://zh-hant-hk.com/zh/req_uri/index.php?a=1&b=2#hash'],
 
         # other patterns should be keep original
         ['a=1&b=2', 'zh-Hant-HK', 'a=1&b=2'],
```

Comment:
Do we need to consider special cases like this? Or can we simply treat this as relative URLs, e.g.:
```ruby
http://zh-hant-hk.com/zh/req_uri/a=1&b=2
```

Reply:
well they are actually relative paths, but the expectation is incorrect and they should be treated like any other relative path